hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3553,7 +3553,7 @@ def reshape(self, shape):
         :class:`.NDArrayExpression`.
         """
         if isinstance(shape, TupleExpression):
-            shape_ir = hl.tuple([hl.int64(i) for i in shape])._ir
+            shape_ir = hl.cond(hl.is_defined(shape), hl.tuple([hl.int64(i) for i in shape]), hl.null(hl.ttuple(*[i.dtype for i in shape])))._ir
```

Comment:
This is `hl.or_missing(hl.is_defined(shape), hl.tuple(...))`

Reply:
That's a good one to know about, way shorter